### PR TITLE
added settings for vpn check for network gateways

### PR DESCRIPTION
**Pull request by muzzamilinovaqo**

### CODE REVIEW

File: plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js [hunk 1]
```diff
@@ -11,11 +11,19 @@ module.exports = {
     recommended_action: 'Configure subnets and network security groups instead of virtual network gateways',
     apis: ['resourceGroups:list','virtualNetworks:listAll','virtualNetworkGateways:listByResourceGroup'],
     realtime_triggers: ['microsoftnetwork:virtualnetworks:write','microsoftnetwork:virtualnetworks:delete','microsoftnetwork:virtualnetworkgateways:write','microsoftnetwork:virtualnetworkgateways:delete'],
-
+    settings: {
+        vNet_gatewayType: {
+            name: 'Virtual Network Gateway Type',
+            description: 'Gateway type that should not be configured with virtual networks i.e. Vpn, ExpressRoute. Enter "*" if no network gateways should be used irrespective of gateway type.',
+            regex: '^.*$',
```
Comment (mehakseedat63) [inline plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js:18]:
```suggestion
            regex: '^.*$',
```
regex should only allow vpn, expressroute and * as values

File: plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js [hunk 2]
```diff
@@ -11,11 +11,19 @@ module.exports = {
...
+            default: '*',
```
Comment (mehakseedat63) [inline plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js:19]:
```suggestion
            default: '*',
```
don't think we need * we can just leave blank for any type

File: plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js [hunk 3]
```diff
@@ -87,7 +101,15 @@ module.exports = {
                 if ((gatewayUsed && !subnetFound) || !gatewayUsed) {
                     helpers.addResult(results, 0, 'Virtual network is not using network gateways', location, virtualNetwork.id);
                 } else {
-                    helpers.addResult(results, 2, 'Virtual network is using network gateways', location, virtualNetwork.id);
+                    if (config !== '*') {
+                        if (restrictGateways.length) {
+                            helpers.addResult(results, 2, `Virtual network is using ${config} network gateways: ${restrictGateways.join(',')}`, location, virtualNetwork.id);
```
Comment (mehakseedat63) [inline plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js:106]:
```suggestion
                            helpers.addResult(results, 2, `Virtual network is using following ${config} network gateways: ${restrictGateways.join(',')}`, location, virtualNetwork.id);
```

File: plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js [hunk 4]
```diff
@@ -87,7 +101,15 @@ module.exports = {
...
+                        } else {
+                            helpers.addResult(results, 0, `Virtual network is not using ${config} network gateways`, location, virtualNetwork.id);
```
Comment (mehakseedat63) [inline plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js:108]:
```suggestion
                            helpers.addResult(results, 0, `Virtual network is not using any ${config} network gateways`, location, virtualNetwork.id);
```

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js [hunk 5]
```diff
@@ -58,25 +66,31 @@ module.exports = {
                 
                 if (virtualNetworkGateways.data.length) {
                     for (let virtualNetworkGateway of virtualNetworkGateways.data) {
-                        gatewaysList.push(virtualNetworkGateway.id);
+                        gatewaysList.push({id:virtualNetworkGateway.id, name: virtualNetworkGateway.name, gatewayType: virtualNetworkGateway.gatewayType});
```
Comment (alphadev4) [inline plugins/azure/virtualnetworks/noNetworkGatewaysInUse.js:69]:
```suggestion
                        gatewaysList.push({id:virtualNetworkGateway.id, name: virtualNetworkGateway.name, gatewayType: virtualNetworkGateway.gatewayType? virtualNetworkGateway.gatewayType: ''});
```

**Review by alphadev4 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**